### PR TITLE
http: added support for %VIRTUAL_CLUSTER_NAME% to extract the matched VC name as a custom header

### PR DESCRIPTION
**Pull request by agrawroh**
This PR adds a new command operator in the header formatter called `%VIRTUAL_CLUSTER_NAME%` which can be used to extract the Virtual Cluster name that gets matched (if any).

**Commit Message:** Added support for `%VIRTUAL_CLUSTER_NAME%` to extract the matched VC name as a custom header.
**Additional Description:** Adds a new command operator called `%VIRTUAL_CLUSTER_NAME%` in the http header formatter which can be used to extract the matched Virtual Cluster name as a custom header.
**Risk Level:** Low
**Testing:** Unit Tests
**Docs Changes:** Added description on `%VIRTUAL_CLUSTER_NAME%` in the docs.
**Release Notes:** Added
**Platform Specific Features:** N/A

**Signed-off-by:** Rohit Agrawal <rohit.agrawal@databricks.com>

### CODE REVIEW

**Comment by agrawroh** [comment]
@zuercher Could you please take a look at this? It adds the same VC command operator for the http header.

File: source/common/router/header_formatter.cc [hunk 1]
```diff
@@ -245,6 +245,14 @@ StreamInfoHeaderFormatter::StreamInfoHeaderFormatter(absl::string_view field_nam
     field_extractor_ = [](const StreamInfo::StreamInfo& stream_info) -> std::string {
       return std::string(stream_info.downstreamAddressProvider().requestedServerName());
     };
+  } else if (field_name == "VIRTUAL_CLUSTER_NAME") {
+    field_extractor_ = [](const Envoy::StreamInfo::StreamInfo& stream_info) -> std::string {
+      absl::optional<std::string> virtual_cluster_name = stream_info.virtualClusterName();
```
Comment (jmarantz) [inline source/common/router/header_formatter.cc:250]: can you try replacing the body of this lambda with:
```
return stream_info.virtualClusterName().value_or("");
```
I think that should work, though maybe you might need
```
return stream_info.virtualClusterName().value_or(std::string(""));
```

See https://github.com/abseil/abseil-cpp/blob/9336be04a242237cd41a525bedfcf3be1bb55377/absl/types/optional.h#L503

**Review by jmarantz — COMMENTED** [review]
generally looks great with one minor suggestion. try that out, in the meantime I'll kick to senior maintainers.

File: source/common/router/header_formatter.cc (hunk 1 above)
Comment (agrawroh) [reply source/common/router/header_formatter.cc]: Done, Thank You!

**Review by jmarantz — APPROVED** [review]

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**